Extract log-based replication test into a different file and convert it to jest

liranmauda opened the pull request:
### Explain the changes
1. Extract log-based replication test into a different file and convert it to jest

### Testing Instructions:
1.  Run `npx jest --testPathPattern="test_bucket_log_based_replication.test.js"`

- [x] Tests added

liranmauda commented:
Draft as it is pending on #7412

Neon-White reviewed (commented):
No major comments :) looks good

liranmauda committed:
Extract log-based replication test into a different file and convert it to jest

Extract log-based replication test into a different file and convert it to jest

Signed-off-by: liranmauda <liran.mauda@gmail.com>
